src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -1136,7 +1136,7 @@ void sendUnsub(@NonNull NatsSubscription sub, int after) {
         if (after > 0) {
             bab.append(SP).append(after);
         }
-        queueOutgoing(new ProtocolMessage(bab, true));
+        queueOutgoing(new ProtocolMessage(bab));

```

Comment:
The true used to mean filter out when the writer is stop (it calls pause then filter), There was one case, the CONNECT message that used false. But I'm certain this was wrong and now all protocol messages are filtered out of the queue on writer stop. More in another comment.